File: bench/wc/wc.lua
-- $Id: wc.lua,v 1.1 2004-05-19 18:13:52 bfulgham Exp $
-- http://www.bagley.org/~doug/shootout/
-- from Roberto Ierusalimschy

BUFSIZE = 2^12

local cc,lc,wc = 0,0,0
while 1 do
    local lines, rest = io.read(BUFSIZE, "*l")
    if lines == nil then break end
    if rest then lines = lines..rest..'\n' end
    cc = cc+string.len(lines)
    local _,t = string.gsub(lines, "%S+", "")   -- count words in the line
    wc = wc+t
    _,t = string.gsub(lines, "\n", "\n")   -- count newlines in the line
    lc = lc+t
end

io.write(lc, " ", wc, " ", cc, "\n")

File: bench/wc/wc.php
#!/usr/bin/php -f
<?php
/*
 $Id: wc.php,v 1.2 2004-06-12 16:19:44 bfulgham Exp $
 http://www.bagley.org/~doug/shootout/
 
 TBD - this program should not assume lines are less than 10000 characters long
*/

$fd = fopen("php://stdin", "r");
$nl = $nw = $nc = 0;
while (!feof ($fd)) {
    if ($line = fgets($fd, 10000)) {
	++$nl;
	$nc += strlen($line);
	$nw += count(preg_split("/\s+/", $line, -1, PREG_SPLIT_NO_EMPTY));
    }
}
fclose($fd);
print "$nl $nw $nc\n";
?>
